contracts/rewards/StandardRewards.sol

```diff
@@ -739,7 +735,21 @@ contract StandardRewards is IStandardRewards, ReentrancyGuardUpgradeable, Utils,
             }
 
             ClaimData memory claimData = _claimRewards(provider, p);
-            rewardData.amount += claimData.reward;
+
+            if (claimData.reward > 0) {
+                uint256 remainingRewards = p.remainingRewards;
+
+                // a sanity check that the reward amount doesn't exceed the remaining rewards per program
+                if (remainingRewards < claimData.reward) {
```

Comment:
this isn't needed, the following subtraction ensures that it'll revert - is that here just for the error message?
if so, I wouldn't bother, since it's an edge case anyway, not an error that anyone should ever see unless there's a bug :)

Reply:
Yeah, I wanted to have an explicit revert reason if this ever happens (not that it should)

Reply:
👍

Reply:
I'm updating the revert reason, though.